bundle-core/src/main/java/org/whispersystems/libsignal/SessionCipher.java

```diff
@@ -0,0 +1,573 @@
+        this(store, store, store, store, remoteAddress);
+    }
+    /**
+     * THIS METHOD ASSUMES IDENTITY IS TRUSTED
+     * USE THE METHOD WITH STREAMING, ENCRYPTION FOR TESTING
+     *
+     * @param  paddedMessage The plaintext message bytes, optionally padded to a constant multiple.
+     * @return A ciphertext message encrypted to the recipient+device tuple.
+     */
+    private CiphertextMessage encrypt(byte[] paddedMessage) throws UntrustedIdentityException {
+        synchronized (SESSION_LOCK) {
+            SessionRecord sessionRecord   = sessionStore.loadSession(remoteAddress);
+            SessionState  sessionState    = sessionRecord.getSessionState();
+            ChainKey      chainKey        = sessionState.getSenderChainKey();
+            MessageKeys   messageKeys     = chainKey.getMessageKeys();
+            ECPublicKey   senderEphemeral = sessionState.getSenderRatchetKey();
+            int           previousCounter = sessionState.getPreviousCounter();
+            int           sessionVersion  = sessionState.getSessionVersion();
+
+            byte[]            ciphertextBody    = getCiphertext(messageKeys, paddedMessage);
+            CiphertextMessage ciphertextMessage = new SignalMessage(sessionVersion, messageKeys.getMacKey(),
+                                                                    senderEphemeral, chainKey.getIndex(),
+                                                                    previousCounter, ciphertextBody,
+                                                                    sessionState.getLocalIdentityKey(),
+                                                                    sessionState.getRemoteIdentityKey());
+
+            if (sessionState.hasUnacknowledgedPreKeyMessage()) {
+                UnacknowledgedPreKeyMessageItems items = sessionState.getUnacknowledgedPreKeyMessageItems();
+                int localRegistrationId = sessionState.getLocalRegistrationId();
+
+                ciphertextMessage = new PreKeySignalMessage(sessionVersion, localRegistrationId, items.getPreKeyId(),
+                                                            items.getSignedPreKeyId(), items.getBaseKey(),
+                                                            sessionState.getLocalIdentityKey(),
+                                                            (SignalMessage) ciphertextMessage);
+            }
+
+            sessionState.setSenderChainKey(chainKey.getNextChainKey());
+
+
+            //identityKeyStore.saveIdentity(remoteAddress, sessionState.getRemoteIdentityKey());
+            //sessionStore.storeSession(remoteAddress, sessionRecord);
+            return ciphertextMessage;
+        }
+    }
+
+    /**
+     * Encrypt a message.
+     *
+     * @return A ciphertext message encrypted to the recipient+device tuple.
+     */
+    public CiphertextMessage encrypt(InputStream inputStream, OutputStream outputStream) throws IOException {
```

Comment:
i think we want to keep the original (non-streaming) encrypt as well to test with.